Client/Ecosia/UI/MyImpact/MyImpactStackView.swift

```diff
@@ -0,0 +1,235 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+@objc protocol MyImpactStackViewModelResize: AnyObject {
+    @objc func resizeStack(sender: MyImpactStackView)
+}
+
+struct MyImpactStackViewModel {
+    let title: String
+    let highlight: Bool
+    let subtitle: String?
+    let imageName: String
+
+    var callout: Callout?
+
+    struct Callout {
+        var action: Action
+        var collapsed: Bool = true
+    }
+
+    enum Action {
+        case tap(text: String, action: Selector)
+        case collapse(text: String, button: String, action: Selector)
+
+        var selector: Selector {
+            switch self {
+            case .tap(_, let action):
```

Comment:
This could be simplified as 1 case:
```suggestion
            case .tap(_, let action), .collapse(_ , _, let action):
```

Reply:
that's convenient! I haven't known this syntax